ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -139,6 +154,29 @@ public void run() {
   }
 
   /**
+   * Pauses hard deletes. If hard deletes is half way in hard deleting a range, it will continue to finish hard deleting
+   * the range and then pause. Hard delete will be in paused state until {@link #resume()} is called.
+   */
+  void pause() throws InterruptedException {
+    reentrantLock.lock();
+    if (!paused.getAndSet(true)) {
```

Comment:
this should be wrapped in a `try..finally` and the unlock should be in `finally`.